pycolmap/examples/write_pose_priors_to_database.py

```diff
@@ -0,0 +1,156 @@
+):
+    """
+    Update the pose prior for a specific image name in the database.
+    If the pose prior doesn't exist, insert a new one.
+
+    Args:
+        colmap_db (pycolmap.Database): colmap database to update.
+        image_name (str): name of the image to update.
+        position (np.array): Position as a 3-element array (x, y, z).
+        coordinate_system (int): Coordinate system index (default: -1).
+        position_covariance (np.array): 3x3 position covariance matrix
+                                        (default: None).
+    """
+    # Get image_id from image_name
+    if colmap_db.exists_image(image_name):
+        position = np.asarray(position, dtype=np.float64).reshape(3, 1)
+        if position_covariance is None:
+            position_covariance = np.full((3, 3), np.nan, dtype=np.float64)
+        image = colmap_db.read_image(image_name)
+        if colmap_db.exists_pose_prior(image.image_id):
+            colmap_db.update_pose_prior(
+                image.image_id,
+                pycolmap.PosePrior(
+                    position,
+                    position_covariance,
+                    pycolmap.PosePriorCoordinateSystem(coordinate_system),
+                ),
+            )
+        else:
+            colmap_db.write_pose_prior(
+                image.image_id,
+                pycolmap.PosePrior(
+                    position,
+                    position_covariance,
+                    pycolmap.PosePriorCoordinateSystem(coordinate_system),
+                ),
+            )
+    else:
+        logging.warning(f"Image at path {image_name} not found in database.")
+
+
+def write_pose_priors_to_database():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--database_path", type=str, required=True)
+    parser.add_argument("--pose_priors_path", type=str, required=True)
+    parser.add_argument("--pose_priors_delimiter", type=str, default=" ")
+    parser.add_argument(
+        "--coordinate_system",
+        type=int,
+        default=0,
+        help="(-1: unknwon, 0: WGS84, 1: Cartesian)",
+    )
+    parser.add_argument(
+        "--use_covariance_from_pose_priors_file",
+        type=bool,
+        default=False,
+        help="If False, use prior_position_std options to set a \
+            common covariance to the priors.",
+    )
+    parser.add_argument("--prior_position_std_x", type=float, default=1.0)
+    parser.add_argument("--prior_position_std_y", type=float, default=1.0)
+    parser.add_argument("--prior_position_std_z", type=float, default=1.0)
+    args = parser.parse_args()
+
+    database_path = Path(args.database_path)
+    pose_priors_path = Path(args.pose_priors_path)
+
+    if not database_path.exists():
+        print("ERROR: database path does not exist.")
+        return
+
+    if not pose_priors_path.exists():
+        print("ERROR: pose priors path already does not exist.")
+        return
+
+    colmap_db = pycolmap.Database(database_path)
```

Comment:
nit: the recommended way is to use pycolmap's new context manager as:
```
with pycolmap.Database(database_path) as colmap_db:
```
as this guarantees that the database is closed even if an exception occurs in between.